src/Fmt_ast.ml

```diff
@@ -1553,15 +1557,21 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext
             $ fmt_expression c (sub_exp ~ctx e2) )
         $ fmt_atrs )
   | Pexp_tuple es ->
-      let parens =
+      let no_parens_if_break =
         match xexp.ctx with
-        | Str {pstr_desc= Pstr_eval _} -> false
-        | _ -> true
+        | Exp {pexp_desc= Pexp_extension _} -> true
+        | Pld _ -> true
+        | Str {pstr_desc= Pstr_eval _} -> true
+        | _ -> false
+      in
+      let wrap =
+        if parens || Poly.(c.Conf.parens_tuple = `Always) then
```

Comment:
Nit: other functions have a type constraint on the `Conf.t` argument when needed and use type-based disambiguation.

Reply:
This changes the behavior in `Always` mode, before the `let parens` disables them for `Pstr_eval`.